### PR TITLE
new: organize gradle dependencies, update Kotlin and Gradle dependencies

### PR DESCRIPTION
**Pull request by Wyvest**
## Description
Organizes all our dependencies to either settings.gradle.kts (for PGT) and libs.version.toml (for everything else)
Updates a lot (a lot) of dependencies

## Checklist
- [x] I made a clear description of what was changed
- [x] I stated why these changes were necessary
- [x] I updated documentation or said what needs to be updated
- [x] I made sure these changes are backwards compatible
- [x] This pull request is for one feature/bug fix

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -79,49 +79,45 @@ dependencies {
...
     compileOnly("org.jetbrains:annotations:23.0.0")
 
-    compileOnly("gg.essential:vigilance-1.8.9-forge:+") {
+    compileOnly(libs.vigilance) {
         attributes { attribute(common, true) }
         isTransitive = false
     }
 
-    compileOnly("cc.polyfrost:universalcraft-1.8.9-forge:246") {
+    compileOnly("cc.polyfrost:universalcraft-1.8.9-forge:${libs.versions.universalcraft.get()}") {
```
Comment (xtrm-en) [inline build.gradle.kts:87]: shouldn't it just be a library and not a version? like `compileOnly(libs.universalcraft)`

File: build.gradle.kts [hunk 2]
```diff
@@ -79,49 +79,45 @@ dependencies {
...
+    shade(libs.kotlinStdlibJdk7)
+    shade(libs.kotlinReflect)
+
+    shade(libs.kotlinxCoroutinesCore)
+    shade(libs.kotlinxCoroutinesCoreJvm)
+    shade(libs.kotlinxCoroutinesJdk8)
+    shade(libs.kotlinxSerializationCore)
+    shade(libs.kotlinxSerializationJson)
+    shade(libs.kotlinxSerializationCbor)
+    shade(libs.kotlinxAtomicfu)
```
Comment (xtrm-en) [inline build.gradle.kts:115]: Please use dependency bundles like `libs.bundle.kotlin` ([cool guide](https://proandroiddev.com/gradle-version-catalogs-for-an-awesome-dependency-management-f2ba700ff894))

File: build.gradle.kts [hunk 3]
```diff
@@ -79,49 +79,45 @@ dependencies {
...
+
+    shade(libs.mixin) {
         isTransitive = false
     }
-    compileOnly("cc.polyfrost:lwjgl-legacy:1.0.0-alpha24") {
+    compileOnly("cc.polyfrost:lwjgl-legacy:${libs.versions.lwjgl.get()}") {
```
Comment (xtrm-en) [inline build.gradle.kts:120]: Same as universalcraft

File: versions/root.gradle.kts [hunk 4]
```diff
@@ -1,10 +1,10 @@
+@file:Suppress("DSL_SCOPE_VIOLATION")
 /**
  * WHATEVER HAPPENS HERE MUST ALSO BE APPLIED IN lwjgl-repacked
  */
 plugins {
-    kotlin("jvm") version "1.6.21" apply false
-    id("cc.polyfrost.multi-version.root")
-    id("io.github.juuxel.loom-quiltflower-mini") version "171a6e2e49" apply false
+    alias(libs.plugins.kotlin) apply false
+    id(pgtLibs.plugins.pgtRoot.get().pluginId)
```
Comment (xtrm-en) [inline versions/root.gradle.kts:7]: `.get().pluginId` ??? shouldn't it work with just `pgtLibs.plugins.pgtRoot`?

**Review by xtrm-en — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (Wyvest) [reply build.gradle.kts:87]: we need to do this for the platform buildscript so it doesn't matter

**Review by xtrm-en — APPROVED** [review]
lgtm

**Review by pauliesnug — DISMISSED** [review]
ily

**Comment by Wyvest** [comment]
Should be ready for reviews

File: versions/src/main/java/cc/polyfrost/oneconfig/internal/renderer/LwjglManagerImpl.java [hunk 5]
```diff
@@ -357,7 +357,7 @@ private static synchronized URL getJarFile() {
         ;
         //#if MC>11202 && MC<=11900
         //$$ String osArch = System.getProperty("os.arch");
-        //$$ jarName = jarName + "-" + ((osArch.startsWith("arm") || osArch.startsWith("aarch64")) ? "arm" : "noarm");
+        //$$ jarName = jarName + "-" + ((osArch.startsWith("arm") || osArch.startsWith("aarch64") || !osArch.contains("64")) ? "arm" : "noarm");
```
Comment (nxtdaydelivery) [inline versions/src/main/java/cc/polyfrost/oneconfig/internal/renderer/LwjglManagerImpl.java:360]: ?

Comment (Wyvest) [reply versions/src/main/java/cc/polyfrost/oneconfig/internal/renderer/LwjglManagerImpl.java:360]: for 32-bit systems - its handled the same as ARM and in the same jar

**Review by nxtdaydelivery — APPROVED** [review]
lgtm

**Pull request merged**